### PR TITLE
refactor(outputs.azure_data_explorer): Changes corresponding to new ADX SDK Standards

### PR DESCRIPTION
**Pull request by asaharn**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

 Changes as per the new Azure data explorer SDK. Modified test cases accordingly.

### CODE REVIEW

**Comment by asaharn** [comment]
@srebhan  Have raise a new PR. Please take a look and suggest If there are any changes required. 
Note: This is the new PR we have raise as per the discussion on #11587

**Comment by powersj** [comment]
Hi @asaharn,

Thanks for clarifying. I have closed the old PR and requested @srebhan review this PR. He is currently out, but when he gets back can get around to this.

Thanks!

**Comment by asaharn** [comment]
Hi @powersj , thanks for update. 

Hi @srebhan , Please have a look whenever you get back and get time. That would be really helpful.

**Comment by asaharn** [comment]
Hi @srebhan , hoping that you are back from vacations. Let me know if something else is required for the changes.

File: plugins/outputs/azure_data_explorer/azure_data_explorer.go [hunk 1]
```diff
@@ -75,22 +65,38 @@ func (adx *AzureDataExplorer) Connect() error {
...
+	adx.ingestors = nil
 
-	return nil
+	if len(errs) == 0 {
+		adx.Log.Info("Closed ingestors and client")
+		return nil
+	}
+
+	// Combine errors into a single object and return the combined error
+	return kustoerrors.GetCombinedError(errs...)
```
Comment (srebhan) [inline plugins/outputs/azure_data_explorer/azure_data_explorer.go:99]: Can we please invert this
```suggestion
	if len(errs) > 0 {
		// Combine errors into a single object and return the combined error
		return kustoerrors.GetCombinedError(errs...)
	}
	
	adx.Log.Info("Closed ingestors and client")
	return nil
```
This makes it easier to extend that function later and also eases following the flow.

File: plugins/outputs/azure_data_explorer/azure_data_explorer.go [hunk 2]
```diff
@@ -156,7 +162,8 @@ func (adx *AzureDataExplorer) pushMetrics(ctx context.Context, format ingest.Fil
 	if err != nil {
 		return err
 	}
-
+	length := len(metricsArray)
+	adx.Log.Debugf("Metrics array length %d for table %s", length, tableName)
```
Comment (srebhan) [inline plugins/outputs/azure_data_explorer/azure_data_explorer.go:166]: This is very technical. How about
```suggestion
	adx.Log.Debugf("Writing %s metrics to table %q", length, tableName)
```

File: plugins/outputs/azure_data_explorer/azure_data_explorer_test.go [hunk 3]
```diff
@@ -18,217 +17,138 @@ import (
...
+	mockClient := kusto.NewMockClient()
+	expectedResultMap := map[string]string{metricName: `{"fields":{"value":1},"name":"test1","tags":{"tag1":"value1"},"timestamp":1257894000}`}
+	mockMetrics := testutil.MockMetrics()
+	// Multi tables
+	mockMetrics2 := testutil.TestMetric(1.0, "test2")
+	mockMetrics3 := testutil.TestMetric(2.0, "test3")
+	mockMetricsMulti := make([]telegraf.Metric, 2)
+	mockMetricsMulti[0] = mockMetrics2
+	mockMetricsMulti[1] = mockMetrics3
+	expectedResultMap2 := map[string]string{"test2": `{"fields":{"value":1.0},"name":"test2","tags":{"tag1":"value1"},"timestamp":1257894000}`, "test3": `{"fields":{"value":2.0},"name":"test3","tags":{"tag1":"value1"},"timestamp":1257894000}`}
```
Comment (srebhan) [inline plugins/outputs/azure_data_explorer/azure_data_explorer_test.go:31]: Can we please change the formatting here so it is easier to see what is going on?
```suggestion
	mockClient := kusto.NewMockClient()
	expectedResultMap := map[string]string{
		"test1": `{"fields":{"value":1},"name":"test1","tags":{"tag1":"value1"},"timestamp":1257894000}`},
	}
	mockMetrics := testutil.MockMetrics()
	// Multi tables
	mockMetricsMulti := []telegraf.Metric{
		testutil.TestMetric(1.0, "test2"),
		testutil.TestMetric(2.0, "test3"),
	}
	expectedResultMap2 := map[string]string{
		"test2": `{"fields":{"value":1.0},"name":"test2","tags":{"tag1":"value1"},"timestamp":1257894000}`,
		"test3": `{"fields":{"value":2.0},"name":"test3","tags":{"tag1":"value1"},"timestamp":1257894000}`,
	}
```

**Review by srebhan — CHANGES_REQUESTED** [review]
Thanks @asaharn for the update! I have some smaller comment in the code.

However, I still think that you should more strictly split the changes to the code and the test. Please change `azure_data_explorer.go` as you did (plus my comments) and keep changes to the test to a **bare minimum**. That is, change the mock client stuff but *DO NOT* reorganize the test code or change the way results are checked!
The reorganization of tests can (and should) be done in a separate PR *without* changing the code.

**Comment by asaharn** [comment]
Closing PR, #11874 raised.

**Pull request closed without merging**